src/renderer/components/SketchParams/SketchParams.tsx

```diff
@@ -30,8 +32,11 @@ const getInputElement = (valueType: NodeTypes, id: string) => {
 }
 
 const ParamItem = ({ param: { key, title, id, valueType } }: ParamProps) => {
+  //const onSelectNode = useOnSelectNode(id)
+  const onSelectNode = useEngineStore((state) => state.updateSelectedNode).bind(null, id)
+  const selected = useEngineStore((state) => state.selectedNode)
   return (
-    <NodeControl key={key}>
+    <NodeControl key={key} onClick={onSelectNode} isActive={id === selected}>
```

Comment:
I was originally following the flow of value updates, going through 2 classes in `src/renderer/components/hooks/` (see below)

It appears that I can just hit the state directly via useEngineStore, do we actually want/need these extra files? Do we want to simplify the node value updating too?

Reply:
So my philosophy is to try and reduce the complexity of components as much as possible by creating hooks. The components just get their functionality from hooks that are consuming state. This means the components don't really need to care about what's going on behind the scenes. We could even swap out the whole state management system and the components wouldn't ever need updating.

However I do realise there's a bit too much boilerplate here. We don't need `useUpdateSelectNode`, that could just be implemented inside of `useOnSelectNode`.